src/VisualStudio/Core/Def/TableDataSource/Suppression/VisualStudioSuppressionFixService.cs

```diff
@@ -157,38 +148,38 @@ private static string GetFixTitle(bool isAddSuppression)
     private static string GetWaitDialogMessage(bool isAddSuppression)
         => isAddSuppression ? ServicesVSResources.Computing_suppressions_fix : ServicesVSResources.Computing_remove_suppressions_fix;
 
-    private IEnumerable<DiagnosticData>? GetDiagnosticsToFix(bool selectedEntriesOnly, bool isAddSuppression)
+    private async Task<ImmutableHashSet<DiagnosticData>?> GetDiagnosticsToFixAsync(
+        bool selectedEntriesOnly,
+        bool isAddSuppression)
     {
         var diagnosticsToFix = ImmutableHashSet<DiagnosticData>.Empty;
-        void computeDiagnosticsToFix(IUIThreadOperationContext context)
-        {
-            var cancellationToken = context.UserCancellationToken;
 
-            // If we are fixing selected diagnostics in error list, then get the diagnostics from error list entry
+        var result = await InvokeWithWaitDialogAsync(async cancellationToken =>
+        {            // If we are fixing selected diagnostics in error list, then get the diagnostics from error list entry
             // snapshots. Otherwise, get all diagnostics from the diagnostic service.
```

Comment:
```suggestion
        {
            // If we are fixing selected diagnostics in error list, then get the diagnostics from error list entry
            // snapshots. Otherwise, get all diagnostics from the diagnostic service.
```